README.md

```diff
@@ -18,6 +18,15 @@ of the pipeline description, and more.
  * **Mailing List**: image-builder@redhat.com
  * **Changelog**: https://github.com/osbuild/osbuild/releases
 
+### Principles
+
+1. [OSBuild stages](./stages) are never broken, only deprecated. The same manifest should always produce the same output.
```

Comment:
I think it would be nice if we revisited this at some point.

I  feel like this principle will just lead to code that is suboptimal. i.e. we're carrying all the old history and never able to give things a proper refactor.

Here's an example of something that is not great..

https://github.com/osbuild/osbuild/blob/a85b51bb2c1af5debf425928f6e7d3b8a603f860/stages/org.osbuild.sgdisk#L33-L37

It makes it so that using the sfdisk versus sgdisk stage isn't consistent. Now, do they really need to be consistent? No. But it's kind of a wart. Wouldn't it be great it we could drop that at some point?

The idea here being that we have spec versions kind of like what [Ignition has](https://coreos.github.io/ignition/specs/). `sgdisk` options at older spec/schema versions would continue to produce the same output, but newer versions could adopt new options/defaults.

Reply:
Yeah, I think that stage versioning is something that we will probably have to do at some point. :) I don't actually think that introducing such a mechanism is against this principle.

Reply:
Yup, we recently discussed stage versioning again. I think we're all open to that and it sounded like it would just happen at some point.